### ISSUE TITLE
sweetAlert reference is not available to itself in a module

### ISSUE THREAD

**Issue by matthew-dean** [t4t5/sweetalert#345]
When referenced as an AMD module, sweetAlert isn't attached to the global scope (which is appropriate). However, this causes internal references to sweetAlert within the library itself to fail, because they aren't passed a reference to the object.

```
ReferenceError: sweetAlert is not defined
    at handleCancel (.../sweetalert.js:428:5)
    at Object.handleButton (.../sweetalert.js:386:9)
```

**Comment by zauni** [comment]
This is explaining the issue in #337. Thanks for this clarification!

**Comment by matthew-dean** [comment]
By the way, this is easily fixed by moving the ```var sweetAlert, swal;``` declaration just inside the outer function, so that the variable is available within the script:
```
;(function(window, document, undefined) {
"use strict";
/*
 * Global sweetAlert function
 * (this is what the user calls)
 */
var sweetAlert, swal;
```

**Comment by wejendorp** [comment]
:+1: nice catch.

**Comment by tonillo** [comment]
i encountered the same problem and i tried to solve this issue as reported below: 
this is my solution for working properly with RequireJS
in sweetalert-dev.js change the RequireJS section from line 261 to 273

```javascript
/*
 * Use SweetAlert with RequireJS
 */
if (typeof define === 'function' && define.amd) {
  define(function () {
    if(typeof window !== 'undefined'){
      window.sweetAlert = window.swal = sweetAlert;
    }
    return sweetAlert;
  });
} else if (typeof module !== 'undefined' && module.exports) {
  module.exports = sweetAlert;
}
```

**Comment by chetankhargone** [comment]
Hi,
I am using sweet-alert and but when I am calling swal function of sweet alert, I am getting error:-
TypeError: swal is not a function
    at sweet-alert.min.js:27

any help highly appreciated.. 
Thank you

**Comment by balogic** [comment]
I am also getting this error :hand:

**Comment by t4t5** [comment]
I believe this is fixed in 2.0. If not, please reopen the issue!
